trackintel/analysis/speed.py

```diff
@@ -0,0 +1,83 @@
+import trackintel as ti
+import numpy as np
+import pandas as pd
+
+
+def speed_positionfixes(positionfixes):
+    """
+    Compute speed per positionfix
+
+    Parameters
+    ----------
+    positionfixes : GeoDataFrame (as trackintel positionfixes)
+        The positionfixes have to follow the standard definition for positionfixes DataFrames.
+
+    Returns
+    -------
+    pfs: GeoDataFrame (as trackintel positionfixes)
+        The original positionfixes with a new column ``[`speed`]``. The speed is given in km/h
+
+    Notes
+    -----
+    The speed at one positionfix is computed from the distance and time since the previous positionfix. For the first
+    positionfix, the speed is set to the same value as for the second one.
+    """
+    pfs = positionfixes.copy()
+    crs_is_projected = ti.geogr.distances.check_gdf_planar(pfs)
+
+    if crs_is_projected:
+        dist_function = lambda x: x.geom.distance(x.prev_geom)
+    else:
+        dist_function = lambda x: ti.geogr.point_distances.haversine_dist(
+            x.geom.x, x.geom.y, x.prev_geom.x, x.prev_geom.y
+        )[0]
```

Comment:
why do we need the [0]? the return of `ti.geogr.point_distances.haversine_dist` is only a float number?

Reply:
As you can see [here](https://github.com/mie-lab/trackintel/blob/3a67d0c7dd2ac1f305b37c81a17e49365d693d79/trackintel/geogr/distances.py#L56), haversine_dist always returns an array.